### PR TITLE
adding docker-compose and jwt support

### PR DESCRIPTION
**Pull request by Amit-limbasiya**

### CODE REVIEW

File: JmeterCLIDemo/JmeterCLIDemo.csproj.user
Comment (goldytech) [inline JmeterCLIDemo/JmeterCLIDemo.csproj.user:1]: csproj.user should not be committed in vcs

File: JmeterCLIDemo/Program.cs [hunk 1]
```diff
@@ -1,19 +1,47 @@
...
+}).AddJwtBearer(x =>
+{
+    x.TokenValidationParameters = new TokenValidationParameters
+    {
+        ValidIssuer = config["JwtSettings:Issuer"],
+        ValidAudience = config["JwtSettings:Audience"],
+        IssuerSigningKey = new SymmetricSecurityKey(Encoding.UTF8.GetBytes(config["JwtSettings:Key"])),
+        ValidateIssuer = true,
+        ValidateAudience= true,
+        ValidateLifetime = false,
```
Comment (goldytech) [inline JmeterCLIDemo/Program.cs:37]: @Amit-limbasiya `ValidateLifeTime` should be set to true. We need to validate the token expiry.

File: data/Entropy.bin
Comment (goldytech) [inline data/Entropy.bin:1]: @Amit-limbasiya Why are the .bin files in VCs ? Which app creates this file

File: data/ODataAPIDatabase.mdf
Comment (goldytech) [inline data/ODataAPIDatabase.mdf:1]: @Amit-limbasiya Sql server db files should not be in VCS ? Give a .SQL script files to create the required schema and insert the data if you are not using the EFCore Migrations. This is not urgent but should be done before the project is closed.

File: docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+version: '3.4'
+
+services:
+  ms-sql-server:
+    image: mcr.microsoft.com/mssql/server:2019-GA-ubuntu-16.04
+    environment:
+        ACCEPT_EULA: "Y"
+        MSSQL_SA_PASSWORD: Password1234
```
Comment (goldytech) [inline docker-compose.yml:8]: @Amit-limbasiya This should come from env vars

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      - "5001:80"
+      - "5000:443"
+
+  jmeter:
+    image: justb4/jmeter:latest 
+    volumes: 
+    - ./test-scripts:/jmeter/test-scripts
+    - ./results:/jmeter/results
+
+    entrypoint: ["sh", "-c","jmeter -n -t \"/jmeter/test-scripts/TestFileDelete.jmx\" -l /jmeter/results/resultsDelete.jtl -f -Jusers=1 -Jrampup=1 -Jendpoint=\"https://jmeterclidemo:443/DeleteByTownCode/823942\" -JhttpMethod=\"DELETE\" -Jresponsetime=3000 -Jloop=100"]
```
Comment (goldytech) [inline docker-compose.yml:31]: @Amit-limbasiya Create a custom .sh file that should execute all jmeter files for the endpoint. This script should be invoked when docker image is run

File: results/resultsDelete.jtl
Comment (goldytech) [inline results/resultsDelete.jtl:1]: @Amit-limbasiya Should this file be part of VCS ?

File: test-scripts/TestFilePut.jmx
Comment (goldytech) [inline test-scripts/TestFilePut.jmx:1]: @Amit-limbasiya Where are we putting the JWT token in the Authorization header in the .jmx file

**Review by goldytech — CHANGES_REQUESTED** [review]
@Amit-limbasiya Work on the given feedback. Let me know if you have any questions.

File: test-scripts/TestFilePut.jmx
Comment (Amit-limbasiya) [reply test-scripts/TestFilePut.jmx:1]: Yes sir, we are putting the JWT token inside the jmx file http header configuration part.

File: results/resultsDelete.jtl
Comment (Amit-limbasiya) [reply results/resultsDelete.jtl:1]: It containes the results of my tests. I think it should not be the part of VCS, I will remove this.

**Pull request merged**